src/instructlab/data/generate.py

```diff
@@ -214,10 +222,27 @@ def generate(
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
         # TODO (cdoern): we really should not edit the cfg object
-        ctx.obj.config.serve.llama_cpp.llm_family = model_family
-        backend_instance = backends.select_backend(
-            cfg=ctx.obj.config.generate.teacher, model_path=model_path
+        gen_cfg = copy.copy(ctx.obj.config)
+        gen_cfg.generate.teacher.llama_cpp.llm_family = model_family
```

Comment:
This should be changed in the `generate.teacher.vllm.llm_family` as well.

Reply:
it wasnt prior, right? would not like to add too many unnecessary changes here

Reply:
That was an oversight and a should have been in. If you don't make the change here, I'll open up a PR for it. We're setting a chat template in vllm too (https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/backends/vllm.py#L269)